.github/workflows/sarplus.yml

```diff
@@ -0,0 +1,168 @@
+#   * [Azure Databirkcs runtime releases](https://docs.microsoft.com/en-us/azure/databricks/release-notes/runtime/releases)
+#   * Package publish
+#       + [Publishing package distribution releases using GitHub Actions CI/CD workflows](https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/)
+#       + [pypa/gh-action-pypy-publish](https://github.com/pypa/gh-action-pypi-publish)
+
+
+name: sarplus package
+
+on:
+  push:
+    paths:
+      - contrib/sarplus/python/**
+      - contrib/sarplus/scala/**
+      - .github/workflows/sarplus.yml
+
+env:
+  PYTHON_ROOT: ${{ github.workspace }}/contrib/sarplus/python
+  SCALA_ROOT: ${{ github.workspace }}/contrib/sarplus/scala
+  SARPLUS_VERSION: 0.5.0
+
+jobs:
+  python:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10"]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install -U build pip twine
+          python -m pip install -U flake8 pytest pytest-cov scikit-learn
+
+      - name: Lint with flake8
+        run: |
+          cd "${PYTHON_ROOT}"
+          # See https://flake8.pycqa.org/en/latest/user/index.html
+          flake8 .
+
+      - name: Package and check
+        run: |
+          cd "${PYTHON_ROOT}"
+          sed -i -E "s/version[[:space:]]*=[[:space:]]*.*,/version=\"${SARPLUS_VERSION}\",/" setup.py
+          python -m build --sdist
+          python -m twine check dist/*
+
+      - name: Testing
+        env:
+          ACCESS_TOKEN: ${{ secrets.SARPLUS_TESTDATA_ACCESS_TOKEN }}
+        run: |
+          cd "${PYTHON_ROOT}"
+          python -m pip install dist/*.gz
+
+          cd "${SCALA_ROOT}"
+          export SPARK_VERSION=$(python -m pip show pyspark | grep -i version | cut -d ' ' -f 2)
+          SPARK_JAR_DIR=$(python -m pip show pyspark | grep -i location | cut -d ' ' -f2)/pyspark/jars
+          SCALA_JAR=$(ls ${SPARK_JAR_DIR}/scala-library*)
+          HADOOP_JAR=$(ls ${SPARK_JAR_DIR}/hadoop-client-api*)
+          SCALA_VERSION=${SCALA_JAR##*-}
+          export SCALA_VERSION=${SCALA_VERSION%.*}
+          HADOOP_VERSION=${HADOOP_JAR##*-}
+          export HADOOP_VERSION=${HADOOP_VERSION%.*}
+          export VERSION="${SARPLUS_VERSION}"
+          sbt ++"${SCALA_VERSION}"! package
+
+          cd "${PYTHON_ROOT}"
+          pytest --token "${ACCESS_TOKEN}" ./tests
+
+      - name: Upload Python package
+        if: github.ref == 'refs/heads/main' && matrix.python-version == '3.10'
+        uses: actions/upload-artifact@v2
+        with:
+          name: pysarplus-${{ env.SARPLUS_VERSION }}
+          path: ${{ env.PYTHON_ROOT }}/dist/*.gz
+
+      - name: Publish package to PyPI
```

Comment:
This is going to be triggered every time there is a `git push`, isn't it? We probably want to have a separate workflow for publishing onto PyPI.

Reply:
agree, actually publishing to pypi automatically is dangerous, because you can't remove versions, in recommenders we do it manually. LightGBM also does it manually

Reply:
Agree.  I will create a separate workflow triggered manually for publishing.

Reply:
As discussed, publishing will be done manually